### ISSUE TITLE
False positive unresolved var string with extend-type in CLJS

### ISSUE THREAD

**Issue by borkdude** [clj-kondo/clj-kondo#2699]
```
$ clj -M:clj-kondo/dev --lint - --filename foo.cljs <<< "(ns foo (:require [clojure.string :as string])) (defprotocol IDude) (extend-type string  IDude)"
foo.cljs:1:82: warning: Unresolved var: string
```
